server/routes/api.js

```diff
@@ -0,0 +1,98 @@
+const express = require('express')
+const router = express.Router()
+const axios = require('axios')
+const moment = require ("moment")
+const City = require("../model/City")
+
+const apiKey = "f2d2c49051fb06b13f554f1bea9d785e"
+const getUrl = function (cityName) {
+    return `http://api.openweathermap.org/data/2.5/weather?q=${cityName}&units=metric&appid=${apiKey}`
+}
+const cityObj = function (cityData) {
+    let mapedData = {
+        name: cityData.data.name.toLowerCase(),
+        temperature: cityData.data.main.temp,
+        condition: cityData.data.weather[0].description,
+        icon: cityData.data.weather[0].icon,
+        updatedAt:  moment(new Date()).format('LLLL')
+    }
+    return mapedData
+}
+    router.get('/city/:cityName', async function (req, res) {
+        const cityName = req.params.cityName
+        const lon = req.query.lon
+        const lat = req.query.lat
+        if (lon && lat) {
+            try {
+                const weatherData = await axios.get(`http://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&units=metric&appid=${apiKey}`)
+                let mapedData = {
+                    name: weatherData.data.name.toLowerCase(),
+                    temperature: weatherData.data.main.temp,
+                    condition: weatherData.data.weather[0].description,
+                    icon: weatherData.data.weather[0].icon,
+                    lon: weatherData.data.coord.lon,
+                    lat: weatherData.data.coord.lat,
+                    updatedAt: moment(new Date()).format('LLLL')
+
+                }
+                res.send(mapedData)
+
+            } catch (error) {
+                console.log(error)
+            }
+        }
+        try {
+            const weatherData = await axios.get(getUrl(cityName))
+            const getData = cityObj(weatherData)
+            res.send(getData)
+
+        } catch (error) {
+            // console.log(error)
+        }
+    })
+
+    router.get("/cities", async function (req, res) {
+        const cities = await City.find({})
+        cities.forEach(a=>a.saved=true)
```

Comment:
It's a good solution, but remember, only saved documents have an `_id`. So you can always use that as your flag to know if the object is saved or not.